### ISSUE TITLE
Unresponsive script behavior

### ISSUE THREAD

**Issue by damyana79** [annefried/discanno#146]
Hi!
There are two small bugs I came across. I'm adding screenshots, because my explanations ate not the best:
1. When there are passages with too long lines, the preceding text (and annotation) disappears. However, after one scrolls back, the text reappears. This happens for the September_11th_attacks text.

![longlines_1](https://cloud.githubusercontent.com/assets/18260513/14985696/e569d99a-1148-11e6-9f8d-4f35a7f73c20.png)
1. I get an error message about an unresponsive script.
   ![scripterrormessage_2](https://cloud.githubusercontent.com/assets/18260513/14985716/f8a1a948-1148-11e6-9544-13fed3dee2c5.png)
2. After one stops the script, strange "{heading}" block appears in the top right corner.
   ![heading_3](https://cloud.githubusercontent.com/assets/18260513/14985719/ff5a5370-1148-11e6-96b5-94177c6bc117.png)

Kind regards, Damyana

**Comment by JeannedArk** [comment]
To the second bug: Does this bug happen often/ just one time/ in a special case?

**Comment by damyana79** [comment]
It doesn't happen often. It happened after the "unresponsive script" - message which I listed before that (in the second snapshot).

**Comment by JeannedArk** [comment]
How often did the "unresponsive script" message appear?

**Comment by damyana79** [comment]
I think it appears when I open longer texts (September_11th_attacks). I'm going to annotate stuff later today and tomorrow. I'll write again if it appears again or if I notice something else connected to that.

**Comment by damyana79** [comment]
Yes, I tried to finish this text and it appeared three times before I could open the text. Actually, it appears now every time before I try to open a longer text, also with text I haven't started to annotate yet. It finally opens them but then displays the effect on the first snapshot when annotated. Do you have access to the texts? September_11th_attacks, Mars and Germany are such longer texts. You can see the error message when you try to open them.

**Comment by JeannedArk** [comment]
Can you attach these files here? I will have a look

**Comment by damyana79** [comment]
As an annotator I cannot export the files (Actually, that would be a nice feature: exporting pure text files). I don't have project manager account for this project. I can set a new (temporary) password for my annotator account and give it to you, if this would help. Or else you could ask Anne to give you the texts or a project manager/annotator account.

**Comment by JeannedArk** [comment]
I asked @annefried for the raw text files :)

**Comment by damyana79** [comment]
OK :)

**Comment by damyana79** [comment]
Actually, the bug doesn't seem to me that small now. With smaller texts I have no problems at all. But as the annotation of the longer text goes further, it becomes more irritating. As I scroll down, the text above disappears and sometimes I have the situation on the first screenshot but with only a few lines left on the bottom part to be annotated. Or sometimes the text itself doesn't disappear but only the annotation I have just done (if it is also on a span above that just disappeared). So it looks as though I haven't annotated the span yet although I know I have. And the annotation itself becomes slower. The marked spans and the categories appear with a few seconds delay.

**Comment by JeannedArk** [comment]
Let's do not mix the issues. I opened therefore an extra ticket: https://github.com/annefried/discanno/issues/155

**Comment by JeannedArk** [comment]
Note: Create index for some attributes in document.

**Comment by JeannedArk** [comment]
Does this error still happen @damyana79 ?

**Comment by damyana79** [comment]
Hi!
Yesterday I annotated the last long text and it went just perfect. I didn't get the unresponsive script behaviour at all and throughout the annotation the text I wanted to annotate didn't just disappear in front of my eyes as it did before. I still have the feeling, that the reaction time after I selected a span is kind of 2sek longer as it ist with short texts, but it is still OK and much, much better it was before.
